### PR TITLE
Add support for the Resource Introspection endpoint

### PR DESCRIPTION
**Pull request by ahmeroxa**

### CODE REVIEW

File: pkg/meroxa/resource.go [hunk 1]
```diff
@@ -77,6 +77,20 @@ type Resource struct {
 	UpdatedAt   time.Time              `json:"updated_at"`
 }
 
+// ResourceIntrospection represents the introspection results of a Resource
+type ResourceIntrospection struct {
+	ID              int64               `json:"id"`
+	AccountID       int                 `json:"account_id"`
+	UUID            string              `json:"uuid"`
+	ResourceID      int                 `json:"resource_id"`
```
Comment (janelletavares) [inline pkg/meroxa/resource.go:85]: we've been trying not to expose IDs.  there's a lot of entityIdentifier types that are both Name and UUID

File: pkg/meroxa/resource_test.go [hunk 2]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/google/go-cmp/cmp"
```
Comment (janelletavares) [inline pkg/meroxa/resource_test.go:7]: not formatted

File: pkg/meroxa/resource.go [hunk 3]
```diff
@@ -180,6 +194,27 @@ func (c *client) performResourceAction(ctx context.Context, nameOrID string, act
...
+	if err != nil {
+		return nil, err
+	}
+
+	err = handleAPIErrors(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	var ri *ResourceIntrospection
```
Comment (lyuboxa) [inline pkg/meroxa/resource.go:210]: I don't think you want this to be a ptr.

Comment (lyuboxa) [reply pkg/meroxa/resource.go:210]: or maybe you do, but the decode is taking the address of that ptr below?

File: pkg/meroxa/resource.go (hunk 1 above)
Comment (nassor) [reply pkg/meroxa/resource.go:85]: https://github.com/meroxa/platform-api/blob/a9fd56f88a1c9ee916d62f4942c265cf03bef9fd/cmd/api/handlers/resources.go#L132-L136

It seems we only need `nameOrUUID` for the resource.

Comment (ahmeroxa) [reply pkg/meroxa/resource.go:85]: My assumption was that the whole `nameOrUUID` was a stop gap for old endpoints that initially started with ID and that all _new_ endpoints would use UUID exclusively. (Although I'm not sure that's actually what's happening).

In any case, this struct is a reflection of what's returned by the API ¯\_(ツ)_/¯

Comment (janelletavares) [reply pkg/meroxa/resource.go:85]: yeah, to be consistent, the API should do an identifier.NameAndUUID thing rather than ResourceID

**Review by nassor — APPROVED** [review]

**Comment by nassor** [comment]
You probably want to open and close the PR. Something is going on with the GitHub actions that are not running after your first commit.

File: pkg/meroxa/resource.go (hunk 1 above)
Comment (ahmeroxa) [reply pkg/meroxa/resource.go:85]: Given that this is what the API is _currently_ returning, I'm going to go ahead and ship this. We can then loop back and update the API at a later date.

**Pull request merged**